andela/mmdp_site
# MMDP-366 Manage user access on the frontend

**What does this PR do?**

Manages user access on the CMS frontend by protecting authenticated routes and hiding user and group links according to user permissions.

**Description of Task to be completed?**

1. Protect authenticated routes
2. Use interceptors to deal with API errors
3. Ensure redirection after login works properly
4. Hide user and groups sidebar items

**How should this be manually tested?**

1. Fetch and checkout to `ft-frontend-access-control-366`.
2. Logout if you are already logged in.
3. Login with a user that does not have user and groups permissions. The sidebar should not have Users and groups links.
4. Logout and login with a user with user and groups permissions. The sidebar should have Users and groups links.
5. Logout and try to access any route on the application other the login route. You will be redirected to the login page.
6. Log in and change the `userToken` value on localStorage to some invalid value. Try to access an item that requires a user to be logged in. You should be redirected to the login page.

**Any background context you want to provide?**

Any pertinent information that should be considered

**What are the relevant Jira issues?**

[MMDP-366](https://viisaus.atlassian.net/browse/MMDP-366)

**Screenshots (if appropriate)**
##### A user without user and groups permissions

![screenshot 2019-02-22 at 14 17 23](https://user-images.githubusercontent.com/8180548/53239296-b9bda880-36ac-11e9-8e6a-12b460389fbd.png)

##### A user with user and groups permissions

![screenshot 2019-02-22 at 14 24 14](https://user-images.githubusercontent.com/8180548/53239688-ab23c100-36ad-11e9-964c-e884aa6d9864.png)

**Questions:**
N/A
